app/api/v2/handlers/operation_api.py

```diff
@@ -47,3 +73,24 @@ async def get_operation_report(self, request: web.Request):
         access = await self.get_request_permissions(request)
         report = await self._api_manager.get_operation_report(operation_id, access)
         return web.json_response(report)
+
+    '''Overridden Methods'''
+    async def create_object(self, request: web.Request):
+        data = await request.json()
+        await self._error_if_object_with_id_exists(data.get(self.id_property))
+        access = await self.get_request_permissions(request)
+        return await self._api_manager.create_object_from_schema(self.schema, data, access)
+
+    async def create_or_update_object(self, request: web.Request):
+        data, access, obj_id, query, search = await self._parse_common_data_from_request(request)
+        matched_obj = self._api_manager.find_object(self.ram_key, query)
+        if matched_obj and matched_obj.access not in access['access']:
+            raise JsonHttpForbidden(f'Cannot update {self.description} due to insufficient permissions: {obj_id}')
+        return await self._api_manager.create_object_from_schema(self.schema, data, access, matched_obj)
+
+    async def update_object(self, request: web.Request):
+        data, access, obj_id, query, search = await self._parse_common_data_from_request(request)
+        obj = await self._api_manager.find_and_update_object(self.ram_key, data, search)
```

Comment:
should this also be access-restricted?

Reply:
So this method is essentially the same as the base_object_api.py method, it just has to make an asynchronous call. The user is only able to retrieve objects they have access to with this method; if the object isn't included in their access permissions, it simply says "object not found."

Reply:
Okay great, just making sure we do keep access restrictions.